src/datachain/lib/dc.py

```diff
@@ -1128,9 +1130,13 @@ def group_by(
             signal_columns.append(column)
             schema_fields[col_name] = func.get_result_type(self.signals_schema)
 
+        signal_schema = SignalSchema(schema_fields)
```

Comment:
Q: (trying to understand this a bit better)

Why don't we apply `to_partial` to `schema_fields`? can one of the `schema_fields` be similar in a sense that we'll need to preserve a partial model?

Reply:
This is because `schema_fields` here is:

```
schema_fields: dict[str, DataType]
```

In case of partial signals (let's say, `file.path`) it will be: `schema_fields = {"file.path": str}`. We can use `{"file": File}`, for example, but we still need to know partial signal names to preserve them.

That's why we keep simple signals in `schema_fields` and store partial signals in `keep_columns` and creating new schema from existing one using `to_partial`.

Reply:
`{"file.path": str}`

does it mean that partial File won't be created and preserved, though?

I still don't quite understand why for some columns (defined as names) we preserve and construct partials, for other (defined as Func) we don't 🤔